rfc/rfc-61/rfc-61.md

```diff
@@ -0,0 +1,240 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+# RFC-61: Snapshot view management
+
+
+## Proposers
+
+- @<proposer1 @fengjian428>
+
+## Approvers
+ - @<approver1 @xushiyan>
+ - @<approver2 @codope>
+
+## Status
+
+JIRA: [HUDI-4677](https://issues.apache.org/jira/browse/HUDI-4677)
+
+> Please keep the status updated in `rfc/README.md`.
+
+## Abstract
+
+For the snapshot view scenario, Hudi already provides two key features to support it:
+* Time travel: user provides a timestamp to query a specific snapshot view of a Hudi table
+* Savepoint/restore: "savepoint" saves the table as of the commit time so that it lets you restore the table to this savepoint at a later point in time if need be.
+but in this case, the user usually uses this to prevent cleaning snapshot view at a specific timestamp, hence, only clean unused files
+The situation is there some inconvenience for users if they use them directly
+
+Usually users incline to use a meaningful name instead of querying Hudi table with a timestamp, using the timestamp in SQL may lead to the wrong snapshot view being used. 
+for example, we can announce that a new tag of hudi table with table_nameYYYYMMDD was released, then the user can use this new table name to query.
+Savepoint is not designed for this "snapshot view" scenario in the beginning, it is designed for disaster recovery. 
+let's say a new snapshot view will be created every day, and it has 7 days retention, we should support lifecycle management on top of it.
+What this RFC plan to do is to let Hudi support release a snapshot view and lifecycle management out-of-box.
+
+## Background
+Introduce any much background context which is relevant or necessary to understand the feature and design choices.
+typical scenarios and benefits of snapshot view:
+1. Basic idea:
+![basic_design](basic_design.png)
+
+Create Snapshot view based on Hudi Savepoint
+    * Create Snapshot views periodically by time(date time/processing time)
+    * Use External Metastore(such as HMS) to store external view 
+
+Build periodic snapshots based on the time period required by the user
+These Shapshots are stored as partitions in the metadata management system
+Users can easily use SQL to access this data in Flink Spark or Presto.
+Because the data store is complete and has no merged details,
+So the data itself is to support the full amount of data calculation, also support incremental processing
+
+2. Compare to Hive solution
+![resource_usage](resrouce_usage.png)
+
+The Snapshot view is created based on Hudi Savepoint, which significantly reduces the data storage space of some large tables
+    * The space usage becomes (1 + (t-1) * p)/t
+    * Incremental use reduces the amount of data involved in the calculation
+
+When using snapshot view storage, for some scenarios where the proportion of changing data is not large, a better storage saving effect will be achieved
```

Comment:
savepoint commit will record all base files at that point of time and those files will be retained in the hudi table. so it's still the full data at that point. what storage saving is this compared against?

Reply:
if some base files have not changed between two savepoints, these two savepoints can share base files instead of retain two full data